site/_posts/2020-05-27-CSI-integration.md

```diff
@@ -0,0 +1,201 @@
+1. Install Velero with CSI snapshotting feature enabled.
+1. Deploy `csi-app`: a pod using a CSI backed PVC that we will backup and restore.
+1. Use Velero to backup and restore the `csi-app`.
+
+## Set up a CSI environment.
+
+As the [Kubernetes CSI Snapshot Beta API](https://kubernetes.io/docs/concepts/storage/volume-snapshots/) is available starting from Kubernetes `1.17`, your will need to be running Kubernetes `1.17` or later.
+
+This post uses an AKS cluster running Kubernetes `1.17`, with Azure disk CSI driver as an example.
+
+Following instructions to install the Azure disk CSI driver from [here](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/master/docs/install-csi-driver-master.md) run the below command
+
+```bash 
+curl -skSL https://raw.githubusercontent.com/kubernetes-sigs/azuredisk-csi-driver/master/deploy/install-driver.sh | bash -s master snapshot -- 
+```
+
+This script will deploy the following CSI components, CRDs, and necessary RBAC:
+
+- [`deployment.apps/csi-azuredisk-controller`](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/master/deploy/csi-azuredisk-controller.yaml)
+- [`daemonset.apps/csi-azuredisk-node`](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/master/deploy/csi-azuredisk-node.yaml)
+- [`deployment.apps/csi-snapshot-controller`](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/master/deploy/csi-snapshot-controller.yaml)
+
+## Install Velero with CSI support enabled
+
+The CSI volume snapshot capability is currently, as of version 1.4, a beta feature behind the `EnableCSI` feature flag and is not enabled by default.
+
+Following instructions from our [docs website](https://velero.io/docs/csi/), install Velero with the [velero-plugin-for-csi](https://github.com/vmware-tanzu/velero-plugin-for-csi) and using the Azure Blob Store as our BackupStorageLocation. Please refer to our [velero-plugin-for-microsoft-azure documentation](https://github.com/vmware-tanzu/velero-plugin-for-microsoft-azure) for instructions on setting up the BackupStorageLocation. Please note that the BackupStorageLocation should be set up before installing Velero.
+
+Install Velero by running the below command
+
+```bash
+velero install \ 
+--provider azure \ 
+--plugins velero/velero-plugin-for-microsoft-azure:v1.1.0,velero/velero-plugin-for-csi:v0.1.1 \ 
+--bucket $BLOB_CONTAINER \ 
+--secret-file <PATH_TO_CREDS_FILE>/aks-creds \ 
+--backup-location-config resourceGroup=$AZURE_BACKUP_RESOURCE_GROUP,storageAccount=$AZURE_STORAGE_ACCOUNT_ID,subscriptionId=$AZURE_BACKUP_SUBSCRIPTION_ID \ 
+--snapshot-location-config apiTimeout=5m,resourceGroup=$AZURE_BACKUP_RESOURCE_GROUP,subscriptionId=$AZURE_BACKUP_SUBSCRIPTION_ID \ 
+--image velero/velero:v1.4.0 \ 
+--features=EnableCSI 
+```
+
+NOTE: Please use the latest released version of the container images from the [official Velero dockerhub](https://hub.docker.com/u/velero).
+
+## Deploy a CSI application
```

Comment:
As you're deploying a stateful app with CSI-enabled backend storage, not a CSI application, can we reword this segment and the text below?